src/dmd/parse.d

```diff
@@ -2365,8 +2365,11 @@ final class Parser(AST) : Lexer
             check(TOK.rightParentheses);
 
             stc = parsePostfix(stc, &udas);
-            if (stc & AST.STC.static_)
-                error(loc, "postblit cannot be static");
+            if (stc & AST.STC.immutable_ || stc & AST.STC.shared_ ||
+                stc & AST.STC.const_ || stc & AST.STC.static_)
+            {
+                error(loc, "postblit cannot be `const`/`immutable`/`shared`/`static`");
```

Comment:
* Should probably be a depreciation
* Specifying the exact attribute used in the source code in the message would be nicer

Reply:
It doesn't have to be a deprecation since marking a postblit as `const/immutable/shared` makes it impossible to call the copy constructor, so no code can be broken whatsoever.

Reply:
Except it only happen if the code is called. If the postblit is not called, you're breaking the build.

Reply:
Indeed, but having a const/immutable postblit which is not called is in itself an indication of design issues. Anyway, I don't mind making it a deprecation, rather I honestly think it's useless.

Reply:
This is an accept-invalid bug, and virtually any occurrence of the syntax indicates a bug. I'll allow without a deprecation cycle. @MartinNowak agree?

Reply:
I don't see the harm in deprecating it, even if it's an accept-invalid bug. It's trivial to do and, perhaps, we don't need a full deprecation cycle. How about deprecating it in for one release and make it an error in the next?

What does we gain from _not_ deprecating it?

Reply:
@andralex : That's based on the assumption that the error is beneficial to the user, which often it is not.

Example:
Vibe.d has this code somewhere in a module, on a type that's not used.
1) You make this a deprecation: I get a deprecation from Vibe.d but my project keeps building. By the time the deprecation period ends, the deprecation will be fixed. Everyone is happy.
2) You make it an error: My project does not build anymore because of a code I do not use nor care about. I cannot use this compiler release until Vibe.d gets fixed. In addition, forcing me to upgrade Vibe.d might force me to upgrade other dependencies, if the same philosophy is applied. I spend the whole day upgrading the dependencies of my project, trying to find a compatible subset, because someone decided that "it's a bug anyway".

Reply:
@mathias-lang-sociomantic I very well understand. I just want to avoid bureaucracy unless truly needed.

Reply:
This is a breaking change. We have a process for handling breaking change. Why is that so hard to follow ?